DataSetExplorer/Controllers/Dataset/DataSetController.cs

```diff
@@ -0,0 +1,52 @@
+﻿using DataSetExplorer.Controllers.Dataset.DTOs;
+using DataSetExplorer.DataSetBuilder.Model;
+using FluentResults;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace DataSetExplorer.Controllers.Dataset
+{
+    [Route("api/dataset/")]
+    [ApiController]
+    public class DataSetController : ControllerBase
+    {
+        private readonly string _cloneGitPath = "../../ClonedProjects/";
+
+        private readonly IDataSetCreationService _dataSetCreationService;
+
+        public DataSetController(IDataSetCreationService creationService)
+        {
+            _dataSetCreationService = creationService;
+        }
+
+        [HttpPost]
+        public IActionResult CreateDataSet([FromBody] List<ProjectDTO> projects)
+        {
+            var dataSets = new List<DataSet>();
+            foreach (var project in projects)
+            {
+                var result = _dataSetCreationService.CreateDataSetInDatabase(_cloneGitPath, project.Name, project.Url);
```

Comment:
You'll need an async operation https://www.pluralsight.com/guides/using-task-run-async-await that you don't need to await - you just need to get the created Datasets with the status "Building" saved to the database, while the git clone is being conducted and the Code model is being compiled.

Reply:
This is done in service: https://github.com/Clean-CaDET/platform/blob/feature/web-app-de/DataSetExplorer/DataSetBuilder/DataSetCreationService.cs#L26-L32
Is that a good solution?